libs/@local/hash-graph-sdk/python/scripts/generate_filter.py

```diff
@@ -0,0 +1,446 @@
+"""Generate `graph_sdk.filter.path` from `graph_client.models`.
+
+We need to generate 4 classes:
+
+- DataTypePath
+- PropertyTypePath
+- EntityPath
+- EntityTypePath
+
+To generate them we have a configuration file that is located
+under `graph_sdk/filter/config.json`.
+
+The configuration file is a record with the following fields:
+- `data_type`
+- `property_type`
+- `entity_type`
+- `entity`
+
+Each entry has the following fields:
+- `selector`: Fields that require an intermediary `*` selector, and then continue
+              with another type. It is a record, where the key is the field name
+              and the value is the type to continue with.
+              (one of `data_type`, `property_type`, `entity_type`, `entity`)
+- `wildcard`: Fields that go into untyped territory through the properties selector.
+- `continue`: Fields that do not require an intermediary `*` selector, and then continue
+              with another type. It is a record, where the key is the field name and the
+              value is the type to continue with.
+              (one of `data_type`, `property_type`, `entity_type`, `entity`)
+
+(also notice how all fields are of same length, I don't know what it is, but it always
+gives me a sense of satisfaction when I see that)
+"""
+import ast
+import json
+from enum import Enum
+from pathlib import Path
+from typing import Literal, TypedDict
+
+from graph_client.models import (
+    DataTypeQueryToken,
+    EntityQueryToken,
+    EntityTypeQueryToken,
+    PropertyTypeQueryToken,
+)
+
+DIRECTORY = Path(__file__).parent.parent
+
+OntologyType = Literal["data_type", "property_type", "entity_type", "entity"]
```

Comment:
Entities are not an ontology type.

Reply:
renamed in [a8024f2](https://github.com/hashintel/hash/pull/2878/commits/a8024f21f21d70c37d56c608e3d2e6c9f1452796)